File: apis/python_interface_helpers/stk_data_flow.py
from stk_data_flow import *
class stk_data_flow:
	obj_map = {}
	@classmethod
	def type(cls,df):
		# Not sure why I can't use stk_get_data_flow_type() here
		return stk_dftype_ulong(df)
	@classmethod
	def find(cls,dfid):
		if dfid in cls.obj_map:
			return cls.obj_map[dfid]
	@classmethod
	def map(cls,dfid,obj):
		cls.obj_map[dfid] = obj
	@classmethod
	def unmap(cls,dfid):
		cls.obj_map[dfid] = None

File: apis/ruby_api/stk_data_flow.rb
require 'stkdata_flow'
class Stk_data_flow
	@@obj_map = {}
	def self.type(df)
		# Not sure why I can't use stk_get_data_flow_type() here
		Stkdata_flow.stk_dftype_ulong(df)
	end
	def self.find(dfid)
		return @@obj_map[dfid]
	end
	def self.map(dfid,obj)
		@@obj_map[dfid] = obj
	end
	def self.unmap(dfid)
		@@obj_map[dfid] = nil
	end
end
